docs/dev/guidelines/server.rst

```diff
@@ -0,0 +1,148 @@
+
+1. Naming convention:
+=====================
+
+All variables, methods and classes should use CamelCase style. The only difference: a first letter of class should be capital. Most importantly use intention-revealing, pronounceable names.
+
+2. Single responsibility principle:
+===================================
+
+One method should be responsible only for one action, it should do it well and do it only. Reduce coupling, if our method does two or three different things at a time then we should consider splitting the functionality.
+
+3. Small methods:
+=================
+
+There is not a standard pattern for method length among the developers. Someone can say 5, in some cases even 20 lines of code is okay. Just try to make methods as small as possible.
+
+4. Duplication:
+===============
+
+Avoid code duplication. If we cannot reuse a method in other place then probably this method is bad and we should consider a better way to write this method. Use Abstraction to abstract common things in one place.
+
+5. Variables and methods declaration:
+=================================
+
+* Encapsulate the code you feel might changed in future.
+* Make variable and methods private by default and increasing access step by step like from a private to protected and not public.
+* Classes, methods or functions should be Open for extension and Closed for modification(Open Closed Design Principle).
+* Program for the interface and not for implementation, you should use interface type on variables, return types of a method or argument type of methods. Just like using SuperClass type to store object rather using SubClass.
+* The use of interface is to facilitate polymorphism, client should not implement an interface method if its not needed.
+
+6. Structure your code correctly:
+=================================
+
+* Default packages not allowed. It can cause particular problems for Spring Boot applications that use the ``@ComponentScan``, ``@EntityScan`` or ``@SpringBootApplication`` annotations since every class from every jar is read.
+* All variables in the class should be declared at the top of the class.
+* If variable is used only in one method then it would be better to declare it as a local variable of this method.
+* Methods should be declared in the same order as they are used (from top to bottom).
+* More important methods should be declared at the top of a class and minor methods at the end.
+
+7. Database:
+============
+
+* Write performant queries that can also deal with more 1000 objects in a reasonable time
+* Prefer one query that fetches additional data instead of many small queries, but don't overdo it. A good rule of thumb is to query not more than 3 associations at the same time.
+* Think about lazy vs. eager fetching when modeling the data types
+* Simple datatypes: immediately think about whether null should be supported as additional state or not. In most cases it is preferable to avoid null
```

Comment:
```suggestion
* Simple datatypes: immediately think about whether null should be supported as additional state or not. In most cases it is preferable to avoid null.
```